rfp-proposal/identity-directory.md

```diff
@@ -0,0 +1,104 @@
+
+Note: **if a user is not connecting to an archive node, archive data should be unavailable**. The plugins (default, optional, and third party plug-ins) are responsible for realizing this and reporting to the user that some info won't be available if the user connects to only a full node.
+
+### Layouts and Linking
+
+The individual events and positions in the various columns should be **linkable**, and URLs will dictate which columns are visible and in which order. So a URL like https://polkaperson.com/HviHUSkM5SknXzYuPCSfst3CXK4Yg6SWeroP6TdTZBZJbVT#governance@448217&rmrk@330589, would render a screen on which the governance default plugin would scroll to block 448217, and the RMRK column would scroll to block 330589, and RMRK would come AFTER the governance column in the view. Setting it as `#rmrk@330589&governance@448217` would make rmrk appear BEFORE governance. Including a column/plugin without a block number serves to simply activate it in the UI, e.g. https://polkaperson.com/HviHUSkM5SknXzYuPCSfst3CXK4Yg6SWeroP6TdTZBZJbVT#governance@448217&rmrk@330589 would contain the default basic identity column + governance + rmrk, and https://polkaperson.com/HviHUSkM5SknXzYuPCSfst3CXK4Yg6SWeroP6TdTZBZJbVT#governance@448217&rmrk@330589&identityhistoyt&society&remarks would also activate optional columns which show an account's identity registration history, their [Society](https://guide.kusama.network/docs/en/maintain-guides-society-kusama) participation, and all the `system.remark` calls they issued.
+
+#### Default Plugins
+
+- `basic info`: a column with basic information about an account, similar to the sidebar on Polkadot JS Apps UI. Should discern between registrars - it should list each registrar who verified this identity and the verification level they gave (i.e. KnownGood vs KnownBad etc.)
+- `governance`: a column listing all of an account's governance activity like votes, proposals, marking the times when the account was a council member, etc. It should resemble a vertical timeline, with related events referencing each other, quoted-tweet style. Events should be linkable as decribed above, i.e. `governance@477723`. The column should **clearly** mark when a user was a council member but failed to uphold their duties, i.e. there was a motion but the user did not vote, and other interesting info (i.e. the user did not do ANYTHING the council can do while being a council member).
+- `treasury`: a history of an account's interactions with the treasury - tip proposals and endorsements, treasury proposals and grant wins, votes on TP motions if user was council at the time (and clear marks if the user FAILED to vote on a TP motion during his activity as councilor).
+
+#### Optional Plugins
+
+The off-line version (self-run version) should support simple installation of plug-ins like so:
```

Comment:
plugin search & install functionality could be part of the app (like in Sublime Text or VSCode)

Reply:
Since this is just a web app when compiled, I don't see how that would be a thing since the JS / CSS needs to be rebundled with everything and then loaded on each subsequent refresh / launch / request. Maybe good for a later milestone, as perhaps a packaged Electron app or some other similar unholy abomination

Reply:
sure, not urgent